File: frontend/src/api/geofence.ts
import client from './client';
import type { PageData, Geofence, GeofenceAlarm } from '../types';

export const geofenceApi = {
  // ===== 围栏管理 =====
  list: (params?: Record<string, unknown>) =>
    client.get<PageData<Geofence>>('/geofences', { params }),

  getAll: () =>
    client.get<Geofence[]>('/geofences/all'),

  getById: (id: string) =>
    client.get<Geofence>(`/geofences/${id}`),

  create: (data: Partial<Geofence>) =>
    client.post<Geofence>('/geofences', data),

  update: (id: string, data: Partial<Geofence>) =>
    client.put<Geofence>(`/geofences/${id}`, data),

  delete: (id: string) =>
    client.delete<null>(`/geofences/${id}`),

  // ===== 围栏告警 =====
  listAlarms: (params?: Record<string, unknown>) =>
    client.get<PageData<GeofenceAlarm>>('/geofence-alarms', { params }),

  getRecentAlarms: (limit?: number) =>
    client.get<GeofenceAlarm[]>('/geofence-alarms/recent', { params: { limit } }),

  handleAlarm: (id: string, data: { handle_status: number; remark?: string }) =>
    client.put<null>(`/geofence-alarms/${id}`, data),
};

File: internal/router/geofence.go
package router

import (
	"ccplatform/internal/middleware"

	"github.com/gin-gonic/gin"
)

func registerGeofenceRoutes(rg *gin.RouterGroup, h *Handlers) {
	geofences := rg.Group("/geofences")
	{
		geofences.GET("", h.Geofence.List)
		geofences.GET("/all", h.Geofence.GetAll)
		geofences.POST("", middleware.RBAC("system:config"), h.Geofence.Create)
		geofences.GET("/:id", h.Geofence.GetByID)
		geofences.PUT("/:id", middleware.RBAC("system:config"), h.Geofence.Update)
		geofences.DELETE("/:id", middleware.RBAC("system:config"), h.Geofence.Delete)
	}

	alarms := rg.Group("/geofence-alarms")
	{
		alarms.GET("", h.Geofence.ListAlarms)
		alarms.GET("/recent", h.Geofence.GetRecentAlarms)
		alarms.PUT("/:id", middleware.RBAC("alarm:handle"), h.Geofence.HandleAlarm)
	}
}
